chrome-launcher.ts

```diff
@@ -153,7 +155,18 @@ export class Launcher {
       throw new Error(`Platform ${platform} is not supported`);
     }
 
-    this.userDataDir = this.opts.userDataDir || this.makeTmpDir();
+    if (typeof this.opts.userDataDir === 'boolean') {
```

Comment:
maybe set `this.useDefaultProfile` in the profile?

Then in the constructor it would just be something like
```js
if (typeof this.opts.userDataDir === 'boolean') {
  if (!this.opts.userDataDir) {
    this.useDefaultProfile = true;
  } else {
    throw new Error('userDataDir must be false or a path');
  }
} else {
  this.useDefaultProfile = false;
}
```

That would immediately thrown on an invalid `true` for `userDataDir`, and it means that this line could just be 
`this.userDataDir = this.opts.userDataDir || this.makeTmpDir();`

Reply:
that might be too clever for the typescript compiler, I guess. To go with @patrickhulce's suggestion to keep `this.userDataDir` an optional string, in the constructor could do

```js
if (typeof this.opts.userDataDir === 'boolean') {
  if (!this.opts.userDataDir) {
    this.useDefaultProfile = true;
    this.userDataDir = undefined;
  } else {
    throw new Error('userDataDir must be false or a path');
  }
} else {
  this.useDefaultProfile = false;
  this.userDataDir = this.opts.userDataDir;
}
```

the compiler seems to understand `this.opts.userDataDir` would then be `string|undefined` by the last line there.

Then here in `prepare()` it would just be
```js
this.userDataDir = this.userDataDir || this.makeTmpDir();
```

Reply:
Used your latest comment for the change.